## Discontinued!

**The reason** is that the API is limited to a certain variable number of comments per sub-reddit, always bellow 1000. This has been comfirmed for the channel 'funny' which is know to be one of the most popular subreddits.

Many workarounds had been tried in order to surpass this limit, as one of the reasons to choose PRAW was the promise to do so. However, our attemps had not resulted victorious.

# Extract info from the subreddits using PRAW

In this section, we will extract all the submissions and comments for each of the above subreddits. For ease of use, we use Python Reddit API Wrapper (PRAW) module to extract all the comments.

Usage of Reddit API requires authentication through 'Oauth'. An unique user agent is required along with a client id and secret (which is generated when an Reddit application is created).

### 1. Loading

In [1]:
import numpy as np
subReddits = np.loadtxt("Swiss_SR.csv", delimiter=",", dtype=bytes).astype(str)
subReddits = subReddits.tolist()
print(subReddits)

['Switzerland', 'AskSwitzerland', 'Basel', 'Bern', 'BielBienne', 'Buenzli', 'Frauenfeld', 'Fribourg', 'Geneva', 'Liestal', 'Luzern', 'Morcote', 'Neuchatel', 'Schaffhausen', 'SanktGallen', 'Schwiiz', 'Solothurn', 'Stans', 'Suisse', 'Thun', 'Ticino', 'Winterthur', 'Zermatt', 'Zug', 'Zurich', 'Breitling', 'CHTrees', 'FCBasel', 'MatterhornPorn', 'Migros', 'Schweiz', 'SwissArmy', 'SwissArmyKnives', 'SwissBuyers', 'SwissGaming', 'SwissGuns', 'SwissHockey', 'SwissESports', 'SwissMountainDogs', 'SwissNews', 'SwissProblems', 'SwissRap', 'SwissSuperLeague', 'SwissHistory', 'CERN', 'EPFL', 'ETHZ', 'UZH']


### 2. Using the API

In [2]:
# Definitions
my_user_agent = "ada:dvd_ada:v1.0.0 (by /u/dk01reddit)"
my_client_id = ""
my_client_secret = ""
my_username = ""
my_password = ""

In [3]:
import praw
# Create Reddit object
r = praw.Reddit(user_agent=my_user_agent,
                     client_id=my_client_id,
                     client_secret=my_client_secret)

Once authenticated, we can get all the comments for each of the subreddits. Let's try with one of them:

In [8]:
swiss = r.subreddit(subReddits[3])
swiss.display_name

'Bern'

In [9]:
i = 0 #counter submissions
nc = list()
for submission in swiss.new(limit=10000):
    i=i+1
    submission.comments.replace_more(limit=0)
    j = 0 #counter top-lvl-comments
    for top_level_comment in submission.comments:
        j=j+1
        nc.append(j)
print(i)
print(nc)

105
[1, 2, 3, 1, 2, 1, 1, 1, 2, 3, 4, 5, 1, 1, 2, 3, 4, 5, 1, 2, 1, 2, 3, 4, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 1, 2, 3, 1, 2, 3, 4, 1, 2, 3, 1, 1, 1, 2, 1, 1, 2, 3, 4, 5, 6, 1, 1, 2, 3, 1, 1, 2, 3, 1, 2, 3, 1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 1, 1, 2, 3, 4, 5, 1, 1, 2, 3, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 1, 2, 3, 1, 2, 1, 2, 1, 1, 2, 1, 1, 2, 3, 4, 5, 6, 1, 2, 1, 2, 1, 2, 3, 1, 1, 2, 1, 2, 1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 1, 2, 3, 4, 5, 6, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 1, 2, 1, 2, 1, 2, 3, 1, 2, 3, 4, 5, 1, 2, 3, 4, 1, 2, 3, 4, 1, 2]


We find a total of 105 submissions (or posts) in r/Bern, with different number of top-level-comments

### 3. The limit test

In [11]:
#test

fun = r.subreddit('funny')
i=0
for submission in fun.new(limit=5000):
    i=i+1
    
print(i)

871


Indeed, in 'funny' there are many more than that